stylus/base/helpers.styl

```diff
@@ -101,61 +109,71 @@ for $alignment, $text-align in $alignments
 
 for $name, $pair in $colors
   $color = $pair[0]
+
   .has-text-{$name}
     color: $color !important
+
   a.has-text-{$name}
     &:hover
     &:focus
       color: sass-darken($color, 10%) !important
+
   .has-background-{$name}
     background-color: $color
 
 for $name, $shade in $shades
   .has-text-{$name}
     color: $shade !important
+
   .has-background-{$name}
     background-color: $shade !important
 
-.has-text-weight-light
-  font-weight: $weight-light !important
-.has-text-weight-normal
-  font-weight: $weight-normal !important
-.has-text-weight-semibold
-  font-weight: $weight-semibold !important
-.has-text-weight-bold
-  font-weight: $weight-bold !important
+$weights ?= light normal semibold bold
+
+for $weight in $weights
+  .has-text-weight-{$weight}
+    font-weight: $weight !important
```

Comment:
`$weight` meets the literal strings above, use `lookup('$weight-'+$weight)` to relate to pre-defined variables [here](https://github.com/shirohana/bulma.stylus/blob/dev/stylus/utilities/initial-variables.styl#L44-L48).

```stylus
/*
Pre-defined variables:
$weight-light    ?= 300
$weight-normal   ?= 400
$weight-medium   ?= 500
$weight-semibold ?= 600
$weight-bold     ?= 700
*/

$weights = 'light' 'normal' 'medium' 'semibold' 'bold'

for $weight in $weights
  .has-text-weight-{$weight}
    font-weight: lookup('$weight-'+$weight) !important
```

renders correctly

```stylus
.has-text-weight-light {
  font-weight: 300 !important;
}
.has-text-weight-normal {
  font-weight: 400 !important;
}
.has-text-weight-medium {
  font-weight: 500 !important;
}
.has-text-weight-semibold {
  font-weight: 600 !important;
}
.has-text-weight-bold {
  font-weight: 700 !important;
}
```

in your way:

```stylus
.has-text-weight-light {
  font-weight: light !important;
}
.has-text-weight-normal {
  font-weight: normal !important;
}
.has-text-weight-semibold {
  font-weight: semibold !important;
}
.has-text-weight-bold {
  font-weight: bold !important;
}
```